.github/workflows/docker.yml

```diff
@@ -83,4 +83,4 @@ jobs:
         run: >-
           docker run --rm
           recast/recastatlas:sha-${GITHUB_SHA::8}
-          -c "recast --help; recast run testing/busyboxtest --backend local"
+          /bin/bash -c "recast --help; recast run testing/busyboxtest --backend local"
```

Comment:
@lukasheinrich At the moment the Docker build will succeed (though it is clearly working hard to build wheels for all of the Python dependencies) but the run test fails in CI.

```pytb
$ docker run --rm recast/recastatlas:sha-${GITHUB_SHA::8} /bin/bash -c "recast --help; recast run testing/busyboxtest --backend local"
  shell: /usr/bin/bash -e {0}
Usage: recast [OPTIONS] COMMAND [ARGS]...

Options:
  -l, --loglevel TEXT
  --help               Show this message and exit.

Commands:
  auth       Authentication Commands (to gain access to internal data)
  backends   The RECAST computational backends.
  catalogue  The RECAST Analysis Catalogue
  ci         Helper Commands for CI systems
  retrieve   Retrieve RECAST Results from asynchronous submissions
  run        Run a RECAST Workflow synchronously
  software   Build Container Images for RECAST
  status     Get the Status of a asynchronous submission
  submit     Submit a RECAST Workflow asynchronously
  tests      Run a test
2021-09-18 19:08:19,849 | packtivity.asyncback |   INFO | configured pool size to 2
2021-09-18 19:08:20,271 |      yadage.creators |   INFO | no initialization data
2021-09-18 19:08:20,272 |    adage.pollingexec |   INFO | preparing adage coroutine.
2021-09-18 19:08:20,272 |                adage |   INFO | starting state loop.
2021-09-18 19:08:20,404 |     yadage.wflowview |   INFO | added </hello:0|defined|unknown>
2021-09-18 19:08:21,698 |     yadage.wflowview |   INFO | added </world:0|defined|unknown>
2021-09-18 19:08:23,299 |    adage.pollingexec |   INFO | submitting nodes [</hello:0|defined|known>]
2021-09-18 19:08:23,299 |                adage |   INFO | unsubmittable: 0 | submitted: 0 | successful: 0 | failed: 0 | total: 2 | open rules: 0 | applied rules: 2
2021-09-18 19:08:23,301 |      pack.hello.step |   INFO | starting file logging for topic: step
/usr/lib/python3.8/subprocess.py:848: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used
  self.stdout = io.open(c2pread, 'rb', bufsize)
2021-09-18 19:08:24,491 |      pack.hello.step |  ERROR | non-zero return code raising exception
2021-09-18 19:08:24,492 |      pack.hello.step |  ERROR | subprocess failed. code: 1,  command docker pull busybox:latest
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/packtivity/handlers/execution_handlers.py", line 332, in execute_and_tail_subprocess
    raise subprocess.CalledProcessError(
subprocess.CalledProcessError: Command 'docker pull busybox:latest' returned non-zero exit status 1.
2021-09-18 19:08:24,495 |      pack.hello.step |  ERROR | job execution if job {'name': 'hello', 'wflow_node_id': 'eda61574-ed22-498e-8c49-8249ce75a55e', 'wflow_offset': '', 'wflow_stage': 'hello', 'wflow_stage_node_idx': 0, 'wflow_hints': {'is_purepub': False}} raise exception exception
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/packtivity/handlers/execution_handlers.py", line 332, in execute_and_tail_subprocess
    raise subprocess.CalledProcessError(
subprocess.CalledProcessError: Command 'docker pull busybox:latest' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/packtivity/syncbackends.py", line 192, in run_packtivity
    run_in_env(job, env, state, metadata, pack_config, exec_config)
  File "/usr/lib/python3.8/site-packages/packtivity/syncbackends.py", line 130, in run_in_env
    return handler(exec_config, environment, state, job, metadata)
  File "/usr/lib/python3.8/site-packages/packtivity/handlers/execution_handlers.py", line 460, in docker_enc_handler
    result = run(config, state, log, metadata, rspec)
  File "/usr/lib/python3.8/site-packages/packtivity/handlers/execution_handlers.py", line 393, in run_containers_in_docker_runtime
    execute_and_tail_subprocess(
  File "/usr/lib/python3.8/site-packages/packtivity/handlers/execution_handlers.py", line 340, in execute_and_tail_subprocess
    raise RuntimeError("failed container execution subprocess. %s", command_string)
RuntimeError: ('failed container execution subprocess. %s', 'docker pull busybox:latest')
2021-09-18 19:08:24,968 |           adage.node |   INFO | node ready </hello:0|failed|known>
2021-09-18 19:08:24,988 | adage.controllerutil |   INFO | no nodes can be run anymore and no rules are applicable
2021-09-18 19:08:24,989 | adage.controllerutil |   INFO | no nodes can be run anymore and no rules are applicable
2021-09-18 19:08:24,989 |                adage |  ERROR | some weird exception caught in adage process loop
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/adage/__init__.py", line 51, in run_polling_workflow
    for stepnum, controller in enumerate(coroutine):
  File "/usr/lib/python3.8/site-packages/adage/pollingexec.py", line 89, in adage_coroutine
    raise RuntimeError('workflow finished but failed')
RuntimeError: workflow finished but failed
2021-09-18 19:08:24,990 |                adage |  ERROR | node: </hello:0|failed|known> failed. reason: unknown
2021-09-18 19:08:24,990 |                adage |   INFO | unsubmittable: 1 | submitted: 0 | successful: 0 | failed: 1 | total: 2 | open rules: 0 | applied rules: 2
2021-09-18 19:08:32,149 |  yadage.steering_api |   INFO | done. dumping workflow to disk.
2021-09-18 19:08:32,151 | recastatlas.subcomma |  ERROR | caught exception
Traceback (most recent call last):
  File "/code/src/recastatlas/backends/local.py", line 21, in run_workflow
    run_workflow(**spec)
  File "/usr/lib/python3.8/site-packages/yadage/steering_api.py", line 20, in run_workflow
    pass
  File "/usr/lib/python3.8/contextlib.py", line 120, in __exit__
    next(self.gen)
  File "/usr/lib/python3.8/site-packages/yadage/steering_api.py", line 110, in steering_ctx
    execute_steering(
  File "/usr/lib/python3.8/site-packages/yadage/steering_api.py", line 60, in execute_steering
    ys.run_adage(backend)
  File "/usr/lib/python3.8/site-packages/yadage/steering_object.py", line 100, in run_adage
    adage.rundag(controller=self.controller, **self.adage_kwargs)
  File "/usr/lib/python3.8/site-packages/adage/__init__.py", line 137, in rundag
    run_polling_workflow(controller, coroutine, update_interval, trackerlist, maxsteps)
  File "/usr/lib/python3.8/site-packages/adage/__init__.py", line 51, in run_polling_workflow
    for stepnum, controller in enumerate(coroutine):
  File "/usr/lib/python3.8/site-packages/adage/pollingexec.py", line 89, in adage_coroutine
    raise RuntimeError('workflow finished but failed')
RuntimeError: workflow finished but failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/code/src/recastatlas/subcommands/run.py", line 56, in run
    run_sync(name, spec, backend=backend)
  File "/code/src/recastatlas/backends/__init__.py", line 76, in run_sync
    BACKENDS[backend].run_workflow(name, spec)
  File "/code/src/recastatlas/backends/local.py", line 23, in run_workflow
    raise FailedRunException
recastatlas.exceptions.FailedRunException
Error: Workflow failed
Exception ignored in: <function Pool.__del__ at 0x7f5414c59ca0>
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 268, in __del__
    self._change_notifier.put(None)
  File "/usr/lib/python3.8/multiprocessing/queues.py", line 368, in put
    self._writer.send_bytes(obj)
  File "/usr/lib/python3.8/multiprocessing/connection.py", line 200, in send_bytes
    self._send_bytes(m[offset:offset + size])
  File "/usr/lib/python3.8/multiprocessing/connection.py", line 411, in _send_bytes
    self._send(header + buf)
  File "/usr/lib/python3.8/multiprocessing/connection.py", line 368, in _send
    n = write(self._handle, buf)
OSError: [Errno 9] Bad file descriptor
Error: Process completed with exit code 1.
```

So it seems that there are some problems in the `Dockerfile` that will need to be resolved first.

Reply:
Oh I guess

```pytb
subprocess.CalledProcessError: Command 'docker pull busybox:latest' returned non-zero exit status 1.
```

is probably the relevant part here.